download_artifact.py

```diff
@@ -0,0 +1,64 @@
+import json
+import re
+import os.path
+from os.path import expanduser
+
+import requests
+
+import config
+def get_project_name_version(Config):
+    with open("pom.xml") as f:
+        for line in f:
+            if Config.version=="":
+                version = re.search(r"<version>(.*?)(|-SNAPSHOT)</version>", line)
+                if version != None:
+                    Config.version = version.groups()[0]
+                    print(Config.version)
+            if Config.project_name=="":
+                project_name = re.search(r"<name>(.*?)(|-parent)</name>", line)
+                if project_name != None:
+                    Config.project_name = project_name.groups()[0]
+                    print(Config.project_name)
+
+def get_m2_settings(Config):
+    m2_path = os.path.expanduser("~/.m2/settings.xml")
+    with open(m2_path) as f:
+        for line in f:
+            if re.search(r"<username>(.*)</username>", line) != None and Config.username=="":
+                Config.username = re.search(r"<username>(.*)</username>", line).groups()[0]
+            if re.search(r"<password>(.*)</password>", line) != None and Config.password=="":
+                Config.password = re.search(r"<password>(.*)</password>", line).groups()[0]
+
+def get_download_path(Config):
+    latest = 0
+    url = "https://nexus.confluent.io/service/rest/v1/search/assets?q="+Config.project_name+"-package-"+Config.version+"&maven.artifactId="+Config.project_name+"-package"
```

Comment:
Define base url (BASE_URL) as constant.
Prepare the rest of the url using BASE_URL.
Rename url to nexus_url to make it more obvious.
Define "-package" & "-package-" strings as constant as well.

Infact, You can use a class to define this.
Class NexusURL:
   def BASE_URL
   def PACKAGE_LITERAL
   
    # use a function to get the prepared url
   def getNexusURL(project_name, project_version)
        return preparedNexusURL